lapce-app/src/editor/view.rs

```diff
@@ -1268,241 +1424,60 @@ fn get_sticky_header_info(
-    let (cursor_line, _) = view.offset_to_line_col(end);
-
-    let mut renders = Vec::new();
-
-    for line in &screen_lines.lines {
-        let line = *line;
-        if line < start_line {
-            continue;
-        }
-
-        if line > end_line {
-            break;
-        }
-
-        let left_col = match mode {
-            VisualMode::Normal => {
-                if start_line == line {
-                    start_col
-                } else {
-                    0
-                }
-            }
-            VisualMode::Linewise => 0,
-            VisualMode::Blockwise => {
-                let max_col = view.line_end_col(line, false);
-                let left = start_col.min(end_col);
-                if left > max_col {
-                    continue;
-                }
-                left
-            }
-        };
+    let col = col
+        + doc
+            .preedit
+            .with_untracked(|preedit| {
+                preedit.as_ref().and_then(|preedit| {
+                    preedit.cursor.as_ref().and_then(|&(start, _)| {
+                        let preedit_line = doc
+                            .buffer
+                            .with_untracked(|b| b.line_of_offset(preedit.offset));
 
-        let (right_col, line_end) = match mode {
-            VisualMode::Normal => {
-                if line == end_line {
-                    let max_col = view.line_end_col(line, true);
+                        (preedit_line == line).then_some(start)
+                    })
+                })
+            })
+            .unwrap_or(0);
 
-                    let end_offset = view.move_right(
-                        start.max(end),
-                        Mode::Visual(VisualMode::Normal),
-                        1,
-                    );
-                    let (_, end_col) = view.offset_to_line_col(end_offset);
+    if block {
+        let x0 = view.line_point_of_line_col(line, col, FONT_SIZE).x;
 
-                    (end_col.min(max_col), false)
-                } else {
-                    (view.line_end_col(line, true), true)
-                }
-            }
-            VisualMode::Linewise => (view.line_end_col(line, true), true),
-            VisualMode::Blockwise => {
-                let max_col = view.line_end_col(line, true);
-                let right = match horiz.as_ref() {
-                    Some(&ColPosition::End) => max_col,
-                    _ => {
-                        let end_offset = view.move_right(
-                            start.max(end),
-                            Mode::Visual(VisualMode::Normal),
-                            1,
-                        );
-                        let (_, end_col) = view.offset_to_line_col(end_offset);
-                        end_col.max(start_col).min(max_col)
-                    }
-                };
-                (right, false)
-            }
+        let width = if after_last_char {
+            CHAR_WIDTH
```

Comment:
Previously, the default block cursor width was `7.0` and the distance the selection highlight would extend beyond the end of the line when the next line was selected was `7.5`. I've defined `CHAR_WIDTH = 7.5` and use it for both cases.